### ISSUE TITLE
Error trace when KeyCombiner Desktop is launched with an invalid combination set as instant lookup trigger

### ISSUE THREAD

**Issue by laurynas-biveinis** [tkainrad/keycombiner#55]
I've been seeing this error consistently since 0.3.0 or so, on the app startup. After clicking "Ignore" the app appears to work as usual (I only use the training mode, not the background app shortcut lookup)

Error:
```TypeError: Error processing argument at index 0, conversion failure from 
    at /Applications/KeyCombiner.app/Contents/Resources/app.asar/webpack:/src/main/index.js:898:20
    at FSReqCallback.oncomplete (fs.js:160:23)
```
OS: macOS 10.17

**Comment by tkainrad** [comment]
Hi, thanks for reporting this!

I am surprised to hear that. It looks like the app has problems reading or writing the settings file at `/Users/<username>/Library/ApplicationSupport/keycombiner/preferences.json`. The only thing that's in this file is the instant lookup trigger, and whether autostart should be enabled. So, if you don't need to change the defaults there, the error will probably not affect you further.

However, of course, I would very much like to fix this. Could you verify if the file exists? Do you have any idea why KeyCombiner might not be able to read/write there? At the very least, I will try to show a proper error message. If the file does not exist, you could try creating it yourself with the following content:

```json
{"combination":"command+alt+c","autostart":true}
```
This is the default that should be written to the file if it doesn't exist. If writing is the problem, creating it yourself might get rid of the error for you.

**Comment by laurynas-biveinis** [comment]
(I assume `ApplicationSupport` in the path you posted is actually `Application Support`)

The file exists, and appears to be readable/writable:
```
-rw-r--r--    1 laurynas  staff     36 Sep  8 12:01 preferences.json
```

**Comment by tkainrad** [comment]
Thanks for this additional information, I will try to get to the bottom of this. 

If I can't reproduce it, I will likely make a blind solution attempt by switching to a third-party package for settings management instead of trying to do it right myself on all platforms.

One more question, if you don't mind: Is there any additional information about this in the log file at `~/Library/Logs/keycombiner/main.log`?

**Comment by laurynas-biveinis** [comment]
The log has the same error message, one per app startup, but no extra information:
```
...
[2021-11-10 16:20:49.386] [info] Is trusted accessibility client: true
[2021-11-10 16:20:49.398] [info] Has media access to screen (screen recording): denied
[2021-11-10 16:20:49.942] [error] TypeError: Error processing argument at index 0, conversion failure from 
    at /Applications/KeyCombiner.app/Contents/Resources/app.asar/webpack:/src/main/index.js:898:20
    at FSReqCallback.oncomplete (fs.js:160:23)
[2021-11-10 16:20:50.295] [info] Not logged in. Will redirect to https://keycombiner.com/collecting/dashboard/?loggedin
[2021-11-11 16:53:37.736] [info] Is trusted accessibility client: true
[2021-11-11 16:53:37.747] [info] Has media access to screen (screen recording): denied
[2021-11-11 16:53:38.317] [error] TypeError: Error processing argument at index 0, conversion failure from 
    at /Applications/KeyCombiner.app/Contents/Resources/app.asar/webpack:/src/main/index.js:898:20
    at FSReqCallback.oncomplete (fs.js:160:23)
[2021-11-11 16:53:38.733] [info] Not logged in. Will redirect to https://keycombiner.com/collecting/dashboard/?loggedin
[2021-11-12 15:37:07.817] [info] Is trusted accessibility client: true
[2021-11-12 15:37:07.828] [info] Has media access to screen (screen recording): denied
[2021-11-12 15:37:08.417] [error] TypeError: Error processing argument at index 0, conversion failure from 
    at /Applications/KeyCombiner.app/Contents/Resources/app.asar/webpack:/src/main/index.js:898:20
    at FSReqCallback.oncomplete (fs.js:160:23)
[2021-11-12 15:37:08.865] [info] Not logged in. Will redirect to https://keycombiner.com/collecting/dashboard/?loggedin
...
```

**Comment by tkainrad** [comment]
Thanks a lot for this log segment!

This was actually helpful, as it includes the following line:

```
TypeError: Error processing argument at index 0, conversion failure from
```

It seems that KeyCombiner cannot register the instant lookup global shortcut, because whatever it read from `preferences.json` is not valid. Could it be that you set the global lookup trigger to an empty string (because of `from `)?

I suspect that's the case because I can reproduce the exact same error this way. To get rid of the error immediately, you can set any valid combination as an instant lookup trigger via _File->Preferences_. 

If you really don't want to use the instant lookup and also don't want KeyCombiner to register any shortcuts with the system, you can wait for the next version. I will include a check if the combination is an empty string. In that case, KeyCombiner will not register anything and also not show any error. If it's not empty but invalid, KeyCombiner will show a proper error message saying that the lookup trigger could not be registered. Until then, you can just click _Ignore_ on the error which has the same effect.

Thanks again for reporting this!

**Comment by tkainrad** [comment]
A side note: Your report was helpful even beyond this particular issue. @laurynas-biveinis 

Thanks to debugging this issue, I also finally saw why the error message itself was missing from these GitHub issue reports that were triggered from the app.

The first line of the trace is on the same line as the code fences and therefore not rendered/shown 🤦  
This made analyzing several reports very hard in the past. Better late than never, I guess.

**Comment by laurynas-biveinis** [comment]
Heh, I didn't even pay attention that the automatically-generated Github report did not include the actual error message :)

Indeed the key combination is empty, the whole content of the file is
```
{"combination":"","autostart":false}
```
IIRC the default global shortcut conflicted with something, so I tried to disable it this way

**Comment by tkainrad** [comment]
This is now fixed in KeyCombiner Desktop 0.5.0 as described in my previous comment:

- If the combination is set to an empty string, a warning message is printed to the log, but no error dialog is shown, and no global combination is registered.
- If an otherwise invalid combination is set, there is now a proper user-readable error message.

The new version also fixed the GitHub issue submission to properly show the error message. Thanks again for reporting!
